### PR TITLE
fix: allow number type in input

### PR DESCRIPTION
**Pull request by Epictek**
[[中文版模板 / Chinese template](https://github.com/ant-design-blazor/ant-design-blazor/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

#623 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | allow number type in input |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by Epictek** [comment]
Side note, input component could benefit a lot from some unit tests

**Comment by ElderJames** [comment]
I found that when the input value is clear, the bound variable of the number type is not set to 0. Could you fix it?

**Comment by Epictek** [comment]
I'll take a look at it :)

File: components/input/Input.cs [hunk 1]
```diff
@@ -203,6 +206,23 @@ protected virtual async Task OnkeyDownAsync(KeyboardEventArgs args)
             if (OnkeyDown.HasDelegate) await OnkeyDown.InvokeAsync(args);
         }
 
+        protected async Task OnMouseUpAsync(MouseEventArgs args)
+        {
+            if (!EqualityComparer<TValue>.Default.Equals(CurrentValue, _inputValue))
+            {
+                if (!_compositionInputting)
```
Comment (lindexi) [inline components/input/Input.cs:213]: I think we can check the `_compositionInputting` first to improve performance.

我想也许可以先判断 `_compositionInputting` 这个字段，然后再判断 `EqualityComparer<TValue>.Default.Equals(CurrentValue, _inputValue)` 用于提升一点性能，可以认为对 `_compositionInputting` 这个字段判断所需的时间会比 EqualityComparer 的少

而无论是 EqualityComparer 还是 `_compositionInputting` 这个字段 都是实际执行逻辑必须判断的条件，因此调换顺序不会影响逻辑，但是先执行判断速度快的条件可以提升一点性能

**Review by ElderJames — APPROVED** [review]
LGTM

**Pull request merged**